packages/aws-cdk/lib/init-templates/app/dotnet/src/HelloCdk/HelloConstruct.cs

```diff
@@ -0,0 +1,28 @@
+using Amazon.CDK;
+using Amazon.CDK.AWS.IAM;
+using Amazon.CDK.AWS.S3;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace HelloCdk
+{
+    public class HelloConstruct : Construct
+    {
+        private readonly IEnumerable<Bucket> _buckets;
+
+        public HelloConstruct(Construct parent, string id, HelloConstructProps props) : base(parent, id)
+        {
+            _buckets = Enumerable.Range(0, props.BucketCount)
+                .Select(i => new Bucket(this, $"Bucket{i}", new BucketProps()))
```

Comment:
Any reason you need to initialize an empty `BucketProps`? It should be optional I believe

Reply:
It's marked as optional, but there's a bug in `bucket.ts`:

    const encryptionType = props.encryption || BucketEncryption.Unencrypted;

This fails is props is null. This is easy to fix:

    const encryptionType = (props && props.encryption) ?
        props.encryption :
        BucketEncryption.Unencrypted

I'll submit a separate PR with the fix.